src/ServiceControl/Infrastructure/TransportMessageExtensions.cs

```diff
@@ -16,18 +16,20 @@ public static string ProcessingEndpointName(this IReadOnlyDictionary<string, str
                 return endpoint;
             }
 
+            // This message could be a failed message.
+            if (headers.TryGetValue("NServiceBus.FailedQ", out endpoint))
+            {
+                return endpoint;
+            }
+
+            // In v5, a message that comes through the Audit Queue
+            // has it's ReplyToAddress overwritten to match the processing endpoint
             var replyToAddress = headers.ReplyToAddress();
```

Comment:
@mike can make sure that we only use this value in case of *audits*? Maybe to only use this value if no exception headers are present?